### PR TITLE
ColumnMajorMatrix now throws an exception

### LINKED ISSUE #5201: Error handling in ColumnMajorMatrix inverse
Currently when ColumnMajorMatrix calculates an inverse of a singular matrix it just bombs MOOSE out completely.  Could i change this to return an error code instead?

I'd like to use its inverse calculations in my TensorMechanics work, but sometimes the matrices i feed it will be singular (eg, for perfect incompressible plasticity).  I want to handle those cases differently, rather than kill MOOSE completely.

### PR DESCRIPTION
**Pull request by WilkAndy**
when it tries to invert a singular matrix, instead of using mooseError

Fixes #5201

### CODE REVIEW

File: framework/src/utils/ColumnMajorMatrix.C [hunk 1]
```diff
@@ -265,5 +266,5 @@ ColumnMajorMatrix::inverse(ColumnMajorMatrix & invA) const
 #endif
 
   if (return_value)
-    mooseError("error in lapack inverse solve");
+    throw MooseException(return_value);
```
Comment (friedmud) [inline framework/src/utils/ColumnMajorMatrix.C]: We really need to make `mooseException()` quickly before we get to many of these in the code!

@wilkandy : this looks good for now though

**Comment by WilkAndy** [comment]
I'm guessing this Exception stuff is in a state of flux presently.  The above code compiles and tests on my computer, so i'll wait till tomorrow and re-code.

**Comment by jwpeterson** [comment]
> I'm guessing this Exception stuff is in a state of flux presently.

Yes, our plan is to provide a macro, `mooseException`, similar to `mooseError` that throws, both so that users don't have to learn too much about exceptions and so they can be more easily turned off if necessary.

**Comment by permcody** [comment]
@WilkAndy - Derek was just saying that we plan to support a macro like we have for `mooseError` and `mooseWarning` so that all of our reporting mechanisms are consistent. We don't have that interface yet but it's fine. We'll recode as necessary.

**Comment by dschwen** [comment]
So we'll be throwing strings?

**Comment by jwpeterson** [comment]
> So we'll be throwing strings?

We'll be throwing MooseException objects, which are just glorified strings at the moment.

**Comment by WilkAndy** [comment]
thanks everyone,

a

**Comment by permcody** [comment]
Throwing strings is less painful when they hit you. That's why we on the
moose team will definitely be throwing objects... I mean, we really want to
get your attention after all. Muhahaha

**Comment by dschwen** [comment]
> Throwing strings is less painful when they hit you.
Uhm, you may want to revise that statement: http://www.snopes.com/photos/gruesome/disarmed.asp

**Comment by permcody** [comment]
Oh man I clicked on that with my eyes open and everything. Mean!

**Comment by dschwen** [comment]
The **/gruesome/** in the link should have given you a hint...

**Pull request merged**